File: src/main/java/com/example/demo/service/CaliberService.java
package com.example.demo.service;

import com.example.demo.model.dto.DictionaryData;
import com.example.demo.model.dto.DictionaryDataInput;

import java.util.List;

public interface CaliberService {
    DictionaryData addCaliber(DictionaryDataInput dictionaryDataInput);
    List<DictionaryData> findAllCalibers();
    DictionaryData updateCaliber(DictionaryData dictionaryData);
    DictionaryData findCaliberById(Long id);
    void deleteCaliber(Long id);
}

File: WEB/gun-mgmt-app/src/app/services/caliber.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { DictionaryData } from '../types/DictionaryData';
import { DictionaryDataInput } from '../types/DictionaryDataInput';

@Injectable({
  providedIn: 'root'
})
export class CaliberService {
  private apiEndpointUrl = `${environment.apiUrl}/calibers`;

  constructor(private http: HttpClient) { }

  public getAllCalibers() :Observable<DictionaryData[]> {
    return this.http.get<DictionaryData[]>(`${this.apiEndpointUrl}`);
  }

  public getCaliberById(id: number) :Observable<DictionaryData> {
    return this.http.get<DictionaryData>(`${this.apiEndpointUrl}/${id}`);
  }

  public addCaliber(caliberInput: DictionaryDataInput) :Observable<DictionaryData> {
    return this.http.post<DictionaryData>(`${this.apiEndpointUrl}/add`, caliberInput);
  }

  public updateCaliber(caliber: DictionaryData) :Observable<DictionaryData> {
    return this.http.put<DictionaryData>(`${this.apiEndpointUrl}/update`, caliber);
  }

  public deleteCaliber(id: number) :Observable<void> {
    return this.http.delete<void>(`${this.apiEndpointUrl}/delete/${id}`);
  }
}
